sklearn/feature_selection/_rfe.py

```diff
@@ -531,17 +532,14 @@ def predict_log_proba(self, X):
         check_is_fitted(self)
         return self.estimator_.predict_log_proba(self.transform(X))
 
-    def _more_tags(self):
-        tags = {
-            "poor_score": True,
-            "requires_y": True,
-            "allow_nan": True,
-        }
-
-        # Adjust allow_nan if estimator explicitly defines `allow_nan`.
-        if hasattr(self.estimator, "_get_tags"):
-            tags["allow_nan"] = self.estimator._get_tags()["allow_nan"]
-
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        if tags.classifier_tags is not None:
+            tags.classifier_tags.poor_score = True
+        if tags.regressor_tags is not None:
+            tags.regressor_tags.poor_score = True
+        tags.target_tags.required = True
+        tags.input_tags.allow_nan = get_tags(self.estimator).input_tags.allow_nan
```

Comment:
Shouldn't this be `True` by default according to the previous code? I think if the `self.estimator` does not set this tag to `True` it would be `False` by default.

Reply:
previously we'd allow nan if the estimator didn't implement `_get_tags` (which is only true for third party estimators). Now we assume third party estimators don't support nan unless they say so.